### ISSUE TITLE
Defections on joining after start

### ISSUE THREAD

**Issue by wctaiwan** [an-empty-string/codenames#13]
If someone joins a 3 player game after it starts, the existing guesser should defect to one of pink/green, and the new player should become a guesser for the other team. If someone joins a 4+ player game after it starts, they should join one of pink/green such that one team has at most 1 more guesser.

**Comment by an-empty-string** [comment]
> If someone joins a 4+ player game after it starts, they should join one of pink/green such that one team has at most 1 more guesser.

This is implemented already as long as the game was started in VERSUS mode (i.e. it had a pink/blue team).

> If someone joins a 3 player game after it starts, the existing guesser should defect to one of pink/green, and the new player should become a guesser for the other team.

Should this happen unconditionally? I can see games with a large gray team being interesting and would like to keep the possibility there. Here are a few solutions I can see to this:

- Defections happen all the time regardless of game setup
- Defections do not happen if the game was started explicitly in GRAY mode (with !game gray)
- Defections only happen after someone runs !defect or a similar command

Thoughts? I am open to any of these but am leaning towards the second.

**Comment by wctaiwan** [comment]
Yeah I think the second option sounds good. I didn't realize there were explicit modes.
